scripts/spack/configs/blueos_3_ppc64le_ib_p9/spack.yaml

```diff
@@ -0,0 +1,351 @@
+# Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+# Axom Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+
+spack:
+  config:
+    install_tree:
+      root: $spack/..
+      projections:
+        all: '{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+    misc_cache: $spack/../misc_cache
+    test_stage: $spack/../test_stage
+    build_stage::
+    - $spack/../build_stage
+
+  # Regular TPLs do not need views
+  view: false
+
+  compilers::
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags: {}
+      modules: []
+      operating_system: rhel7
+      paths:
+        cc:  /usr/tce/packages/gcc/gcc-8.3.1/bin/gcc
+        cxx: /usr/tce/packages/gcc/gcc-8.3.1/bin/g++
+        f77: /usr/tce/packages/gcc/gcc-8.3.1/bin/gfortran
+        fc:  /usr/tce/packages/gcc/gcc-8.3.1/bin/gfortran
+      spec: gcc@8.3.1.1
+      target: ppc64le
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags: {}
+      modules: []
+      operating_system: rhel7
+      paths:
+        cc:  /usr/tce/packages/gcc/gcc-8.3.1/bin/gcc
+        cxx: /usr/tce/packages/gcc/gcc-8.3.1/bin/g++
+        f77: /usr/tce/packages/gcc/gcc-8.3.1/bin/gfortran
+        fc:  /usr/tce/packages/gcc/gcc-8.3.1/bin/gfortran
+      spec: gcc@8.3.1.2
+      target: ppc64le
```

Comment:
Question for future -- do we still need different entries of the same compiler to support CPU and NVCC builds?

Reply:
I was planning on figuring that out, post this PR. But this PR is huge and I was trying to minimize the changes.